// SPDX-License-Identifier: BSD-3-Clause
// SPDX-FileCopyrightText: 2018-2020 Carles Fernandez-Prades <carles.fernandez@cttc.es>
syntax = "proto3";

package gnss_sdr;

message MonitorGalileoEphemeris {
    /* Galileo ephemeris are 16 parameters and here are reported following the ICD order, paragraph 5.1.1.
       The number in the name after underscore (_1, _2, _3 and so on) refers to the page were we can find that parameter */
    int32 IOD_ephemeris=1;
    int32 IOD_nav_1=2;
    double M0_1=3;            //!< Mean anomaly at reference time [semi-circles]
    double delta_n_3=4;       //!< Mean motion difference from computed value [semi-circles/sec]
    double e_1=5;             //!< Eccentricity
    double A_1=6;             //!< Square root of the semi-major axis [meters^1/2]
    double OMEGA_0_2=7;       //!< Longitude of ascending node of orbital plane at weekly epoch [semi-circles]
    double i_0_2=8;           //!< Inclination angle at reference time  [semi-circles]
    double omega_2=9;         //!< Argument of perigee [semi-circles]
    double OMEGA_dot_3=10;     //!< Rate of right ascension [semi-circles/sec]
    double iDot_2=11;          //!< Rate of inclination angle [semi-circles/sec]
    double C_uc_3=12;          //!< Amplitude of the cosine harmonic correction term to the argument of latitude [radians]
    double C_us_3=13;          //!< Amplitude of the sine harmonic correction term to the argument of latitude [radians]
    double C_rc_3=14;          //!< Amplitude of the cosine harmonic correction term to the orbit radius [meters]
    double C_rs_3=15;          //!< Amplitude of the sine harmonic correction term to the orbit radius [meters]
    double C_ic_4=16;          //!< Amplitude of the cosine harmonic correction term to the angle of inclination [radians]
    double C_is_4=17;          //!< Amplitude of the sine harmonic correction term to the angle of inclination [radians]
    uint32 d_Toe=18;  // Ephemeris reference time

    /*Clock correction parameters*/
    uint32 d_Toc=19;  // Clock correction data reference Time of Week
    double af0_4=20;           //!< SV clock bias correction coefficient [s]
    double af1_4=21;           //!< SV clock drift correction coefficient [s/s]
    double af2_4=22;           //!< SV clock drift rate correction coefficient [s/s^2]

    /*GST*/
    // Not belong to ephemeris set (page 1 to 4)
    int32 WN_5=23;   //!< Week number
    int32 TOW_5=24;  //!< Time of Week
    double Galileo_satClkDrift=25;
    double Galileo_dtr=26;  //!< relativistic clock correction term

    // SV status
    int32 SISA_3=27;
    int32 E5a_HS=28;    //!< E5a Signal Health Status
    int32 E5b_HS_5=29;  //!< E5b Signal Health Status
    int32 E1B_HS_5=30;  //!< E1B Signal Health Status
    bool E5a_DVS=31;      //!< E5a Data Validity Status
    bool E5b_DVS_5=32;    //!< E5b Data Validity Status
    bool E1B_DVS_5=33;    //!< E1B Data Validity Status

    double BGD_E1E5a_5=34;  //!< E1-E5a Broadcast Group Delay [s]
    double BGD_E1E5b_5=35;  //!< E1-E5b Broadcast Group Delay [s]
    
    int32 i_satellite_PRN=36; //!< SV PRN NUMBER
}
